lib/proxy/helpers.rb

```diff
@@ -40,6 +40,25 @@ def https_cert_cn
     end
   end
 
+  # parses the body as json and returns a hash of the body
+  # returns empty hash if there is a json parse error or body is empty
+  def parse_json_body
+    json_data = {}
+    if request.env["CONTENT_TYPE"] == "application/json"
+      begin
+        body_parameters = request.body.read
+        unless body_parameters.nil? || body_parameters.empty?
+          json_data = JSON.parse(body_parameters)
+        end
+      rescue => e
+        logger.debug("Invalid JSON content in body of request: #{e.message}")
+      end
+    else
+      logger.warn("Request header content_type not set to application/json")
```

Comment:
You probably want to notify the caller that you expect json payload in the body.